### DISCUSSION TITLE
Using and controlling soundfonts on GO

### DISCUSSION THREAD

**Discussion by DavidShaw1970** [GrandOrgue/grandorgue#1801] [General]
Is there a way that I can use and control a soundfont on Grandorgue?
Thank you.

**Comment by midimusic** [comment]
By soundfont I assume you mean .sf2 or similar fonts. If so - then no, GO does not recognise these fonts. To use these fonts you need an organ which works with fluidsynth.
csw900

**Comment by DavidShaw1970** [comment]
csw900,

Thank you for answering.
Yes, I do mean .sf2 or similar fonts.
And in reading answer, "then no, GO does not recognise these fonts."

Again, Thank you.

David A. Shaw
________________________________

**Comment by hnb2907** [comment]
Hi @DavidShaw1970 

It's not impossible, but requires some work to convert the contents of the .sf2 to .wav format.  I've recently tried this on a linux based system.  

1. Install package "gigtools", which contains program "sf2extract".  There might be something similar available for Windoze/Mac, sorry I've no idea as I don't have either. ;)
2. Run "sf2extract" on the .sf2 file, which extracts the .wav's  You might need to check them in something like "audacity" to ensure they're in a decent bit and sample rate.  I also used audacity to convert my .wav's from mono to stereo, added some stereo widening, panned the sound away from centre of left-right, and tinkered with some other stuff to clean them up; this is probably optional for you.
3. For me, extracting the .sf2 gave 2x .wav sample files per octave (C# and G#).
4. Then run "LoopAuditioneer" on the .wav files to add the loop and release markers, if required.
5. Next edit your .odf file to add a new [Ranknnn] and its Pipennn's.    
In my case I used the extracted C# .wav files for "Pipennn=_filepath_" relating to B C C# D D# E   -and-   the extracted G# .wav files for pipes relating to F F# G G# A A#.
To get the pipes to speak at the correct semitone, you also add "PipennnPitchTuning=_xxx_" where value _xxx_ is +/-100x cents for each semitone correction required.  For a bit of additional realism, I slightly randomised the +/-100 for each "Pipe_nnn_", maybe any additional 3 or 4, to deliberately make the tuning less perfect.  
Similarly I also randomised each Pipe's gain so they don't all speak at exactly the same volume.  I did this in the "Organ Settings" window, rather than the ODF with "PipennnGain=_yyy_", but either method should work.
 
I think this should get you going :)  It's the first octave and bit.  It shows how to use 3x .wav files and retune them to speak at the correct semitone pitch.
```
[Rank002]
Name=Diapason
FirstMidiNoteNumber=24
NumberOfLogicalPipes=97
WindchestGroup=003
Percussive=N
Gain=12
HarmonicNumber=8

; C from C#
Pipe001=.\Diapason\025-C#.wav
Pipe001PitchTuning=-98

; C# original sample
Pipe002=.\Diapason\025-C#.wav
Pipe002PitchTuning=0

; D from C#
Pipe003=.\Diapason\025-C#.wav
Pipe003PitchTuning=101

; D# from C#
Pipe004=.\Diapason\025-C#.wav
Pipe004PitchTuning=199

; E from C#
Pipe005=.\Diapason\025-C#.wav
Pipe005PitchTuning=300

; F from G#
Pipe006=.\Diapason\032-G#.wav
Pipe006PitchTuning=-303

; F# from G#
Pipe007=.\Diapason\032-G#.wav
Pipe007PitchTuning=-201

; G from G#
Pipe008=.\Diapason\032-G#.wav
Pipe008PitchTuning=-100

; G# original sample
Pipe009=.\Diapason\032-G#.wav
Pipe009PitchTuning=0

; A from G#
Pipe010=.\Diapason\032-G#.wav
Pipe010PitchTuning=99

; A# from G#
Pipe011=.\Diapason\032-G#.wav
Pipe011PitchTuning=200

; B from C#
Pipe012=.\Diapason\037-C#.wav
Pipe012PitchTuning=-203

; C from C#
Pipe013=.\Diapason\037-C#.wav
Pipe013PitchTuning=-101

; C# original sample
Pipe014=.\Diapason\037-C#.wav
Pipe014PitchTuning=0

; D from C#
Pipe015=.\Diapason\037-C#.wav
Pipe015PitchTuning=102

```

A caveat: if the .wav samples contain a tremulant, playing 2+ pipes at the same time, the tremulants will oscillate at obviously different speeds, caused by the "PipennnPitchTuning" differences

If you don't have access to "sf2extract", I don't mind extracting them and providing the .wav's for you, as long as there's no copyright restrictions or similar on the .sf2's.
"LoopAuditioneer" seems to be available for linux/windoze/MAC.  This was the most labourious part of the process, to get the loops sounding correctly.  If you have single-hit percussive instrument samples, you probably don't need to do it.


Enjoy :)
Chris.
